testsuite/testclient/conftest.py

```diff
@@ -0,0 +1,900 @@
+            testcases = yaml.load(fp, Loader=yamlordereddictloader.Loader)
+            for i, testcase in enumerate(testcases):
+                try:
+                    tc_name = testcase['name']
+                except KeyError:
+                    raise ClientTestError("Test case #%s does not have a "
+                                          "'name' attribute" % i + 1)
+                yield YamlItem(tc_name, self, testcase, filepath)
+
+
+class YamlItem(pytest.Item):
+    """
+    py.test test case collector class that runs the test on a single test case.
+    """
+
+    def __init__(self, name, parent, testcase, filepath):
+        super(YamlItem, self).__init__(name, parent)
+        self.testcase = testcase
+        self.filepath = filepath
+
+    def tc_name(self):
+        """
+        Return test case name.
+        """
+        name = self.testcase.get('name', 'unknown')
+        return name
+
+    def runtest(self):
+        runtestcase(self.testcase)
+
+    def repr_failure(self, excinfo):
+        """
+        Called when self.runtest() raises an exception, to provide details
+        about the failure.
+        """
+        exc = excinfo.value
+        if isinstance(exc, ClientTestFailure):
+            return "Failure running test case: %s" % exc
+        elif isinstance(exc, ClientTestError):
+            return "Error in definition of test case: %s" % exc
+        else:
+            return "Error: %s" % exc
+
+    def reportinfo(self):
+        return self.fspath, 0, "%s in %s" % (self.name, self.filepath)
+
+
+class ClientTestError(Exception):
+    """Exception indicating an error in the test case definition."""
+    pass
+
+
+class ClientTestFailure(Exception):
+    """Exception indicating a failure when running a test case."""
+    pass
+
+
+def show_diff(conn, expected, actual, display_text):
+    """Display the actual and expected data"""
+
+    print("Details for the following assertion error:")
+    print("- Expected result %s: %s" % (display_text, expected))
+    print("- Actual result %s: %s" % (display_text, actual))
+    if conn is not None and conn.debug:
+        print("- HTTP response data: %r" % conn.last_raw_reply)
+
+
+def str_tuple(tuple_):
+    """
+    Prepare a tuple or NonType for output.
+
+    This gets around issues of type failure when trying to
+    print tuples.
+
+    Returns str rep of tuple on None if tuple_ is None
+    """
+    return 'NoneType' if tuple_ is None else '%s' % (tuple_,)
+
+
+def obj(value, tc_name):
+    """
+    Create a PyWBEM CIM object or list of CIM Objects from a testcase value.
+
+    If the input value is not a valid type or list of valid types an
+    error is raised.
+
+    If the pywbem_object defines a valid type (ex. CIMInstance) or
+    list of items where the pywbem_object entry defines a valid type,
+    the type is evaluated with the remaining entries in the
+    dictionary and returned.
+
+    Examples for a testcase value (JSON object / Python dict):
```

Comment:
Suggest that this be at the module doc level since it is really the definition of the yaml

Reply:
Added YAML syntax at module level, but this part is for representation of Python values and therefore only a small subset, so I left it here and the new module-level description references it.
DONE.